### DISCUSSION TITLE
Extract and save the SPR key separately

### DISCUSSION THREAD

**Discussion by RMAZANIELLO** [openthread/openthread#9988] [Q&A] [answered]
During the factory reset of the device, we want to erase network data (NVM) to rejoin the network as a new device with new ipv6, (designed successfully) but at the same time, we want to keep SRP configuration (also stored in NVM).

So, would it be possible to provide NVM content specification? 
What do the various fields of NVM represent? 
We want to extract and save the SRP key separately.

Thanks

Regis MAZANIELLO

**Comment by jwhui** [comment, marked as answer]
The documentation in [include/openthread/platform/settings.h](https://github.com/openthread/openthread/blob/main/include/openthread/platform/settings.h) provides details on the information OpenThread stores in non-volatile. Note the keys with prefix `OT_SETTINGS_KEY_SRP_*`.

**Reply by abtink** [reply]
You can also check the [`common/settings.hpp`](https://github.com/openthread/openthread/blob/d099d786201712449d8fff4f489941c87cfeaebe/src/core/common/settings.hpp#L655-L664) definitions (e.g., `SrpEcdsaKey`). Every item saved in non-volatile `Settings` has a corresponding `class/struct` in this header file which specifies its "value" format.
